frontend/src/contexts/ToastContext.tsx

```
@@ -0,0 +1,321 @@
+const generateId = () => `toast-${Date.now()}-${++_idCounter}`;
+
+export const ToastProvider: React.FC<ToastProviderProps> = ({
+  children,
+  config: userConfig,
+}) => {
+  const [toasts, dispatch] = useReducer(reducer, []);
+  const dismissTimers = useRef<Map<string, ReturnType<typeof setTimeout>>>(new Map());
+
+  const config = React.useMemo<Required<GlobalToastConfig>>(
+    () => ({ ...DEFAULT_CONFIG, ...userConfig }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)